### PR TITLE
Add profiles to existing skaffold manifest when initialising with --skaffold

### LINKED ISSUE #250: Add/update skaffold profiles for externally created skaffold manifest file.
When skaffold was created externally i.e. not bootstrapped by Kev, we need to be able to still work with that and inject kev profiles into it. This should allow us to automate some of the dev flows at later stage.  

Need to be mindful of profile names as they may clash with what user already defined so might need to rename profile names generated by Kev to ensure there is no conflicts.

### PR DESCRIPTION
**Pull request by marcinc**
Resolves #250

### CODE REVIEW

File: cmd/kev/cmd/init.go [hunk 1]
```diff
@@ -118,25 +119,43 @@ func runInitCmd(cmd *cobra.Command, _ []string) error {
...
 			})
+
+		} else {
+
+			skaffoldManifest, err := kev.PrepareForSkaffold(envs)
+			if err != nil {
+				return displayInitError(err)
+			}
+
+			if skaffoldManifest != nil {
```
Comment (ezodude) [inline cmd/kev/cmd/init.go:150]: Not sure why this check required, as the lines before already do the work.

File: cmd/kev/cmd/init.go [hunk 2]
```diff
@@ -118,25 +119,43 @@ func runInitCmd(cmd *cobra.Command, _ []string) error {
...
 
-		skaffoldManifest, err := kev.PrepareForSkaffold(manifest, skaffoldManifestPath, envs)
-		if err != nil {
-			return displayInitError(err)
-		}
+		// set skaffold path in kev manifest
+		manifest.Skaffold = kev.SkaffoldFileName
 
-		// don't override existing skaffold.yaml
 		if manifestExistsForPath(skaffoldManifestPath) {
```
Comment (ezodude) [inline cmd/kev/cmd/init.go:125]: So this is a `createOrUpdateSkaffoldManifest` type setup. Refactoring it into a `func` will help make it more obvious.

File: pkg/kev/kev.go [hunk 3]
```diff
@@ -49,15 +48,12 @@ func Init(composeSources, envs []string, workingDir string) (*Manifest, error) {
 }
 
 // PrepareForSkaffold initialises a skaffold manifest for kev project.
-// It'll also set the Skaffold field in kev manifest with skaffold file path passed as argument.
-func PrepareForSkaffold(manifest *Manifest, skaffoldPath string, envs []string) (*SkaffoldManifest, error) {
+func PrepareForSkaffold(envs []string) (*SkaffoldManifest, error) {
```
Comment (ezodude) [inline pkg/kev/kev.go:51]: ❤️ this simplification.

File: pkg/kev/skaffold.go [hunk 4]
```diff
@@ -60,6 +71,22 @@ func LoadSkaffoldManifest(path string) (*SkaffoldManifest, error) {
 	return s, yaml.Unmarshal(data, &s)
 }
 
+// AddProfiles injects kev profiles to existing Skaffold manifest
+// Note, if profile name already exists in the skaffold manifest then profile won't be added
+func AddProfiles(path string, envs []string, includeAdditional bool) (*SkaffoldManifest, error) {
```
Comment (ezodude) [inline pkg/kev/skaffold.go:76]: Looking around the code `includeAdditional` is always `true` ... and `NewSkaffoldManifest`  runs `AdditionalProfiles()`.

Is there a case where we wouldn't want to run `AdditionalProfiles()`?

**Review by ezodude — COMMENTED** [review]
All in all makes sense.

File: pkg/kev/skaffold.go (hunk 4 above)
Comment (marcinc) [reply pkg/kev/skaffold.go:76]: This is useful primarily in testing. Might add additional flag to control whether to include additional (helper) profiles - which currently are included by default - to the manifest in the future PR.

File: cmd/kev/cmd/init.go (hunk 1 above)
Comment (marcinc) [reply cmd/kev/cmd/init.go]: old artefact. removed.

**Review by ezodude — APPROVED** [review]
👍

**Pull request merged**